floodsub_test.go

```diff
@@ -1119,3 +1119,46 @@ func TestMessageSender(t *testing.T) {
 		}
 	}
 }
+
+func TestConfigurableMaxMessageSize(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	hosts := getNetHosts(t, ctx, 10)
+
+	// use a 4mb limit; default is 1mb; we'll test with a 2mb payload.
+	psubs := getPubsubs(ctx, hosts, WithMaxMessageSize(1<<22))
+
+	sparseConnect(t, hosts)
+	time.Sleep(time.Millisecond * 100)
```

Comment:
These make me sad and they're all over this code base. While we could certainly improve the API (e.g. by allowing a Readiness check even without a dummy discovery mechanism) we can already remove the sleep by following the pattern in this test.

https://github.com/libp2p/go-libp2p-pubsub/blob/a1999db8934dc4320aaaa578aed05873113f1a04/topic_test.go#L122

https://github.com/libp2p/go-libp2p-pubsub/blob/a1999db8934dc4320aaaa578aed05873113f1a04/topic_test.go#L145